apps/web/app/(app)/gardens/page.tsx

```diff
@@ -1,68 +1,40 @@
-import React from "react";
+"use client";
+
+import React, { useMemo } from "react";
 import Image from "next/image";
 import { clouds1, clouds2, gardenHeader } from "@/assets";
 import { GardenCard } from "@/components";
 import {
-  getTokenGardensDocument,
   getTokenGardensQuery,
+  getTokenGardensDocument,
 } from "#/subgraph/.graphclient";
-import { initUrqlClient, queryByChain } from "@/providers/urql";
-import {
-  localhost,
-  arbitrumSepolia,
-  optimismSepolia,
-  sepolia,
-} from "viem/chains";
-import { getContractsAddrByChain, isProd } from "@/constants/contracts";
+import useSubgraphQueryMultiChain from "@/hooks/useSubgraphQueryMultiChain";
+import LoadingSpinner from "@/components/LoadingSpinner";
 
 export const dynamic = "force-dynamic";
 
-const { urqlClient } = initUrqlClient();
-
-export default async function Gardens() {
-  const chainsId = [
-    localhost.id,
-    arbitrumSepolia.id,
-    optimismSepolia.id,
-    sepolia.id,
-  ];
-  let gardens: getTokenGardensQuery | null = null;
-  gardens = {
-    tokenGardens: [],
-  };
+export default function Gardens() {
+  const { data: gardens, fetching } =
+    useSubgraphQueryMultiChain<getTokenGardensQuery>(
+      getTokenGardensDocument,
+      {},
+      {},
+      [
+        {
+          topic: "garden",
+        },
+        {
+          topic: "community",
+        },
+      ],
+    );
 
-  try {
-    if (isProd) {
-      const r0 = await getTokenGardens(sepolia.id);
-      gardens?.tokenGardens.push(...r0.data.tokenGardens);
-      const r1 = await getTokenGardens(arbitrumSepolia.id);
-      gardens?.tokenGardens.push(...r1.data.tokenGardens);
-    } else {
-      const promises = [];
-      for (let index = 0; index < chainsId.length; index++) {
-        const chainId = chainsId[index];
-        const subgraph = getContractsAddrByChain(chainId)?.subgraphUrl;
```

Comment:
the logic was moved to some else place, right?

Reply:
Yeah logic is extracted directly inside useSubgraphQueryMultiChain file

But its now a for loop no matter what and the chains list depends on dev or not:
![image](https://github.com/1Hive/gardens-v2/assets/13890618/8536fd9b-23d9-42e3-ae20-81af52c22ffe)